models/src/main/scala/coop/rchain/models/rholangn/Collection.scala

```diff
@@ -78,8 +78,7 @@ object ESetN {
   */
 final class EMapN(val ps: TreeMap[ParN, ParN], val remainder: Option[VarN]) extends CollectionN {
   // Sorted by the hash of the objects which is memoized as part of Rho type
```

Comment:
Comment is a bit misleading, since there is no sorting in the expression.
Maybe smith like this?
```
Content of a map as a sequence sorted by rho type hash of a key . 
Sorting is ensured by constructors in a companion object and TreeMap instance preserving the order on all operations.
```
Btw this class can be instantiated directly without constructor, so someone can make an instance of EMapN sorted in different way. Can we disallow `new EMapN`?

Reply:
I updated the comments.

Yes, I made constructors private because we have apply methods.